component/http/handler.go

```diff
@@ -62,44 +62,77 @@ func determineEncoding(r *http.Request) (string, encoding.DecodeFunc, encoding.E
 	var enc encoding.EncodeFunc
 	var dec encoding.DecodeFunc
 	var ct string
+	var err error
 
 	if cok {
-		switch cth[0] {
-		case "*/*", json.Type, json.TypeCharset:
-			enc = json.Encode
-			dec = json.Decode
-			ct = json.TypeCharset
-		case protobuf.Type, protobuf.TypeGoogle:
-			enc = protobuf.Encode
-			dec = protobuf.Decode
-			ct = protobuf.Type
-		default:
-			return "", nil, nil, errors.New("content type header not supported")
+		ah := getMultiValueHeaders(cth[0])
```

Comment:
Question: Will there be ever more than one content-type? When i am POSTing a request it cannot be both JSON and XML. Maybe this applies only for the accept header...

BTW: the variable ah should be ch for content header.

Reply:
yeah maybe it doesn't make sense for content, I think I added for consistency but it doesn't make a lot of logical sense. 
Alright remove it, added the old code and adjust the logic only for the accept headers.